internal/nix/profiles.go

```diff
@@ -29,47 +30,58 @@ func ProfileList(writer io.Writer, profilePath string, useJSON bool) (string, er
 	return string(out), nil
 }
 
-func ProfileInstall(writer io.Writer, profilePath, installable string) error {
-	if !IsInsecureAllowed() && PackageIsInsecure(installable) {
-		knownVulnerabilities := PackageKnownVulnerabilities(installable)
-		errString := fmt.Sprintf("Package %s is insecure. \n\n", installable)
+type ProfileInstallArgs struct {
+	Installable string
+	Offline     bool
+	ProfilePath string
+	Writer      io.Writer
+}
+
+func ProfileInstall(ctx context.Context, args *ProfileInstallArgs) error {
+	if !IsInsecureAllowed() && PackageIsInsecure(args.Installable) {
+		knownVulnerabilities := PackageKnownVulnerabilities(args.Installable)
+		errString := fmt.Sprintf("Package %s is insecure. \n\n", args.Installable)
 		if len(knownVulnerabilities) > 0 {
 			errString += fmt.Sprintf("Known vulnerabilities: %s \n\n", knownVulnerabilities)
 		}
 		errString += "To override use `devbox add <pkg> --allow-insecure`"
 		return usererr.New(errString)
 	}
 
-	cmd := command(
+	cmd := commandContext(
+		ctx,
 		"profile", "install",
-		"--profile", profilePath,
+		"--profile", args.ProfilePath,
 		"--impure", // for NIXPKGS_ALLOW_UNFREE
 		// Using an arbitrary priority to avoid conflicts with other packages.
 		// Note that this is not really the priority we care about, since we
 		// use the flake.nix to specify the priority.
-		"--priority", nextPriority(profilePath),
-		installable,
+		"--priority", nextPriority(args.ProfilePath),
 	)
+	if args.Offline {
+		cmd.Args = append(cmd.Args, "--offline")
+	}
```

Comment:
I'm not totally familiar with the implications of this.

Reply:
> A ['substituter'](https://nixos.org/manual/nix/unstable/glossary.html#gloss-substituter) in Nix terminology is an upstream server that can serve prebuilt binaries for a given derivation (it can 'substitute' the binary instead of having to build from source).

yes, we are okay disabling them because the `nix print-dev-env` has already evaluated the derivations for the build inputs. And so the binaries are already present on the user's machine.